build-docker-image/action.yml

```diff
@@ -0,0 +1,83 @@
+name: Build docker image
+description: Build a docker image and push it to Github container registry
+
+inputs:
+  docker_repository:
+    description: Repository name in the container registry. e.g. ghcr.io/dfe-digital/register-trainee-teachers
+    required: true
+  github_token:
+    description: Default Github token retrieved via secrets.github_token
+    required: true
+  snyk_token:
+    description: Snyk token for image vulnerability scanning
+  reuse_cache:
+    description: Use previously built docker image layers to improve build time. Set to false to refresh image.
+    default: true
+    type: boolean
```

Comment:
The convention with GitHub Actions (although not written down as far as I can tell, but [is shown in their examples](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#inputs)) seems to be to use dashes in place of underscores with inputs and outputs. Not a blocker, but just thought it was worth sticking to one.